cdap-app-templates/cdap-etl/cdap-data-streams/src/main/java/co/cask/cdap/datastreams/SparkStreamingPipelineDriver.java

```diff
@@ -80,26 +81,54 @@ public void run(final JavaSparkExecutionContext sec) throws Exception {
     boolean checkpointsDisabled = pipelineSpec.isCheckpointsDisabled();
 
     String checkpointDir = null;
+    JavaSparkContext context = null;
     if (!checkpointsDisabled) {
-      // Get the location of the checkpoint directory.
       String pipelineName = sec.getApplicationSpecification().getName();
-      String relativeCheckpointDir = pipelineSpec.getCheckpointDirectory();
-
-      // there isn't any way to instantiate the fileset except in a TxRunnable, so need to use a reference.
-      final AtomicReference<Location> checkpointBaseRef = new AtomicReference<>();
-      Transactionals.execute(sec, new TxRunnable() {
-        @Override
-        public void run(DatasetContext context) throws Exception {
-          FileSet checkpointFileSet = context.getDataset(DataStreamsApp.CHECKPOINT_FILESET);
-          checkpointBaseRef.set(checkpointFileSet.getBaseLocation());
+      String baseCheckpointDir = appendToDirectory(pipelineSpec.getCheckpointDirectory(), pipelineName);
+      checkpointDir = appendToDirectory(baseCheckpointDir, pipelineSpec.getPipelineId());
+      Path checkpointDirPath = new Path(checkpointDir);
+
+      context = new JavaSparkContext();
+      Configuration configuration = context.hadoopConfiguration();
+      // Set the filesystem to whatever the checkpoint directory uses. This is necessary since spark will override
+      // the URI schema with what is set in this config. This needs to happen before StreamingCompat.getOrCreate
+      // is called, since StreamingCompat.getOrCreate will attempt to parse the checkpointDir before calling
+      // context function.
+      configuration.set("fs.defaultFS", checkpointDir);
+      FileSystem fileSystem = FileSystem.get(configuration);
```

Comment:
Does this call error if the checkpointDir is set to something invalid? When a user gives something invalid for the checkpointDir, we should make sure the pipeline fails with a nice error message that mentions that the checkpoint dir is invalid.

Reply:
It will not throw an error here - will attempt to use the default FS "file:///", but will in the first if block below, stating it's unable to create the checkpoint base directory. I think that's a reasonable error message

Reply:
Is there a way we can tell it failed because this is invalid?

We would like to be able to explicitly say that the checkpoint directory is invalid. Otherwise many users will not be able to tell what actions they should take to fix it.

Reply:
Yep, we can create it as a URI. This is also necessary since there's a bug in hadoop.FileSystem